contracts/core/modules/RebalancingTokenIssuanceModule.sol

```diff
@@ -0,0 +1,131 @@
+
+
+/**
+ * @title RebalancingTokenIssuanceModule
+ * @author Set Protocol
+ *
+ * A module that includes a convenience function for redeeming a rebalancing set into components efficiently.
+ */
+contract RebalancingTokenIssuanceModule is
+    ModuleCoreState,
+    ReentrancyGuard
+{
+    using SafeMath for uint256;
+
+    /* ============ Constructor ============ */
+
+    /**
+     * Constructor function for ExchangeIssueModule
+     *
+     * @param _core                The address of Core
+     * @param _transferProxy       The address of transferProxy
+     * @param _vault               The address of Vault
+     */
+    constructor(
+        address _core,
+        address _transferProxy,
+        address _vault
+    )
+        public
+        ModuleCoreState(
+            _core,
+            _transferProxy,
+            _vault
+        )
+    {}
+
+    /* ============ Public Functions ============ */
+
+    /**
+     * Redeems a Rebalancing Set into the base components of the Base Set.
+     *
+     * @param  _rebalancingSetAddress    Address of the rebalancing Set to redeem
+     * @param  _redeemQuantity           The Quantity of the rebalancing Set to redeem
+     */
+    function redeemRBSetIntoBaseComponents(
+        address _rebalancingSetAddress,
+        uint256 _redeemQuantity
+    )
+        public
+        nonReentrant
+    {
+        // Redeem RB Set to the vault attributed to this contract
+        coreInstance.redeemModule(
+            msg.sender,
+            address(this),
+            _rebalancingSetAddress,
+            _redeemQuantity
+        );
+
+        // Calculate the Base Set Redeem quantity
+        IRebalancingSetToken rebalancingSet = IRebalancingSetToken(_rebalancingSetAddress);
+        address baseSetAddress = rebalancingSet.currentSet();
+        uint256 baseSetRedeemQuantity = getBaseSetRedeemQuantity(baseSetAddress);
+
+        // Withdraw base Set to this contract
+        coreInstance.withdraw(
+            baseSetAddress,
+            baseSetRedeemQuantity
+        );
+
+        // Redeem Base Set and send components to the the user
+        coreInstance.redeemAndWithdrawTo(
+            baseSetAddress,
+            msg.sender,
+            baseSetRedeemQuantity,
+            0
+        );
+    }
+
+    /**
+     * Given a rebalancing set address, retrieve the base set quantity redeem quantity.
+     *
+     * @param _baseSetAddress             The address of the base Set
+     * @return baseSetRedeemQuantity      The quantity of base Set to redeem
+     */
+    function getBaseSetRedeemQuantity(
+        address _baseSetAddress
+    )
+        private
+        returns (uint256)
+    {
+        // Get Base Set Details from the rebalancing Set
+        uint256 baseSetNaturalUnit = IRebalancingSetToken(_baseSetAddress).naturalUnit();
+        uint256 baseSetBalance = vaultInstance.getOwnerBalance(
+            _baseSetAddress,
+            address(this)
+        );
+
+        require(
```

Comment:
@asoong Instead of that funky return baseSet, we could do something like a require where the input issue or rebalance quantity needs to work

Reply:
Yeah I like this better, with the tradeoff being issuance quantity granularity?